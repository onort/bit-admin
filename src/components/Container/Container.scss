@import "../../styles/main.scss";

.container {
  flex-grow: 1;
  max-width: px-to-rem(1200px);
  margin: $spacer-md auto;
  padding: 0 $spacer-md;
}

.narrow {
  max-width: px-to-rem(800px);
}
