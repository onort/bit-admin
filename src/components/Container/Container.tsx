import React from "react"
import cx from "classnames"

import styles from "./Container.scss"

interface Props {
  className?: string
  narrow?: boolean
}

const Container: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles.narrow]: props.narrow },
    props.className
  )
  return <div className={className}>{props.children}</div>
}

export default Container
