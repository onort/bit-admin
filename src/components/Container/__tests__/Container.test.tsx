import React from "react"
import { shallow } from "enzyme"

import Container from "../"

const testP = "My test paragraph."
const testClassName = "testClass"

describe("<Container />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Container>
        <p>{testP}</p>
      </Container>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("should have correct css class if narrow is set", () => {
    const wrapper = shallow(
      <Container narrow="narrow">
        <p>{testP}</p>
      </Container>
    )
    expect(wrapper.find("div").hasClass("narrow")).toBe(true)
  })

  it("should be able to have custom classes", () => {
    const wrapper = shallow(
      <Container className={testClassName}>
        <p>{testP}</p>
      </Container>
    )
    expect(wrapper.find("div").hasClass(testClassName)).toBe(true)
  })
})
