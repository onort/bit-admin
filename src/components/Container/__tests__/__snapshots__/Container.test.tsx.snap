// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should match snapshot 1`] = `
<div
  className="container"
>
  <p>
    My test paragraph.
  </p>
</div>
`;
